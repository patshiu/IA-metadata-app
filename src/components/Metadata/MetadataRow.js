import React from 'react';
import PropTypes from 'prop-types';
import './MetadataRow.scss';

const MetadataRow = ({ label, description }) => {
  if (Array.isArray(description)) {
    let parsedDescription = '';
    for (let item of description) {
      parsedDescription += ', ' + item;
    }
    description = parsedDescription.slice(1);
    console.log('description parsed');
  }
  return (
    <div>
      <dt>{label}</dt>
      <dd>{description}</dd>
    </div>
  );
};

MetadataRow.propTypes = {
  label: PropTypes.string.isRequired,
  description: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.array])
};

export default MetadataRow;
