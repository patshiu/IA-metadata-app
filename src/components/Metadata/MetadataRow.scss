@import '../../_variables';

dl {
  div {
    display: flex;
    dt {
        color: $c-medium;
        display: inline-flex;
        width: 20%;
      
        &::after {
          content: ":";
        }
      }
    
      dd {
        color: $c-strong;
        display: inline-flex;
        width: 79%;
        margin: 0;
        padding: 0;
      }
  }
}