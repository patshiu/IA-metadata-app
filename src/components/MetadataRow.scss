
@import '../variables';

dl {
  div {
    display: flex;
    dt {
        color: $c-light;
        display: inline-flex;
        width: 20%;
      
        &::after {
          content: ":";
        }
      }
    
      dd {
        color: $c-medium;
        display: inline-flex;
        width: 79%;
        margin: 0;
        padding: 0;
      }
  }
}