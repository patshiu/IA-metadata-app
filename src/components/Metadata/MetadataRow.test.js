import React from 'react';
import renderer from 'react-test-renderer';

import MetadataRow from './MetadataRow';

test('renders metadata row correcly with string input for description', () => {
  const label = 'subject';
  const description =
    "Futurism;World's Fairs: New York, 1939-40;Infrastructure: Roads and highways";
  const tree = renderer.create(<MetadataRow label={label} description={description} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders metadata row correcly with array input for description', () => {
  const label = 'subject';
  const description = [
    'Futurism',
    "World's Fairs: New York",
    '1939-40',
    'Infrastructure: Roads and highways'
  ];
  const tree = renderer.create(<MetadataRow label={label} description={description} />).toJSON();
  expect(tree).toMatchSnapshot();
});
