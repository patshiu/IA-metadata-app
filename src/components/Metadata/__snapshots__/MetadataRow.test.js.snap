// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders metadata row correcly with array input for description 1`] = `
<div>
  <dt>
    subject
  </dt>
  <dd>
     Futurism, World's Fairs: New York, 1939-40, Infrastructure: Roads and highways
  </dd>
</div>
`;

exports[`renders metadata row correcly with string input for description 1`] = `
<div>
  <dt>
    subject
  </dt>
  <dd>
    Futurism;World's Fairs: New York, 1939-40;Infrastructure: Roads and highways
  </dd>
</div>
`;
